docs/sources/flow/reference/cli/convert.md

```diff
@@ -0,0 +1,35 @@
+---
+title: grafana-agent convert
+weight: 100
+---
+
+# `grafana-agent convert` command
+
+The `grafana-agent convert` command converts a supported configuration file format
+to a Grafana Agent Flow river file.
+
+## Usage
+
+Usage: `grafana-agent convert [FLAG ...] FILE_NAME`
+
+If the `FILE_NAME` argument is not provided or if the `FILE_NAME` argument is
+equal to `-`, `grafana-agent convert` converts the contents of standard input. Otherwise,
+`grafana-agent convert` reads and converts the file from disk specified by the argument.
+
+The `--output` flag can be specified to write the contents of the converted
+file on disk to the specified path.
+
+The `--bypass-warnings` flag can be used to bypass warnings. Warnings are defined as 
+non-blocking issues identified during the conversion for feature[s] on the source file
+not supported in Grafana Agent Flow.
+
+The command fails if the file being converted has syntactically incorrect
+configuration or cannot be converted to a Grafana Agent Flow river file.
+
+The following flags are supported:
+
+* `--output`, `-o`: The filepath and filename where the output is written.
+
+* `--source-format`, `-f`: Required. The format of the source file. Supported formats: 'prometheus'.
```

Comment:
I think we're burying the lede here with a lack of explanation for what `prometheus` here means. 

I'd like to see a section for supported source formats, which go into a little more detail about them. For example, we should document:

* What version of Prometheus is used for conversion, potentially even linking to its documentation
* What broad features are/aren't supported (recording rules, alerting rules, etc)
* Anything else you might think is useful to know about the Prometheus conversion

Reply:
@clayton-cornell what do you think about the best way to organize this information? I was anticipating most of these details would exist in the migration guide for https://github.com/grafana/agent/issues/4073 . If we want to include that work as part of this PR before merge I think that's also fair.

I don't have a strong opinion about the location of this information, but I do agree we want it before I consider the converter 1.0 'done' end to end.

Reply:
Proposal: do we want to make this the staging branch and do the full docs/run CLI changes to this branch then merge it to main all together?

Let's still coalesce on the convert CLI implementation on this PR as written.

Reply:
I'm going to draft some additional docs for this PR and let's go from there

Reply:
@erikbaranowski This is something we've chatted a little about in an offline 1:1 and in messages and it's in-0line with my earlier concerns about this not really providing all info and being confusing to me when I read it. I opted to say "it's follwoing the `fmt` doc, so... I'll leave it as-is for now. It could definitely have a rethink - and if we do document this a bit differently, we should revisit (in another PR?) the `fmt` doc... and maybe `run` as well.

For here, maybe this: follow manpage style for the CLI documentation. Stick to the basics... this is the command and syntax, and here are the arguments you can use... (these are required, these are optional) followed by a common usecase example.

Then for more details on implementation and so on, we shift that to a migration guide that explains the "why woudl I use this" in more detail.

For where to include... keep the PRs simple.  We can keep this for code changes and the basic CLI doc content.  This makes reviews a bit more manageable (bite sized).

Reply:
thanks all, @clayton-cornell can you take 1 more pass on the last commit and see if you want to tune anything for now. I appreciate all the feedback and hope to keep iterating on this.